python/paddle/tensor/math.py

```diff
@@ -8436,3 +8436,59 @@ def isin(
         return cmp.reshape([])
     else:
         return cmp
+
+
+def cartesian_prod(x: Sequence[Tensor], name: str | None = None) -> Tensor:
+    """
+    Perform Cartesian product on a given tensor sequence. This behavior is similar to the itertools.product in Python.
+    Equivalent to converting all input tensors into lists, performing itertools.product on these lists,
+    and finally converting the resulting list into tensors.
+
+    Args:
+        x (list[Tensor]|tuple[Tensor]): Any number of 1-D input Tensors. Supported data types: bfloat16, float16, float32, float64, int32, int64, complex64 or complex128.
+        name (str, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
```

Comment:
```suggestion
        name (str|None, optional): Name for the operation (optional, default is None). For more information, please refer to :ref:`api_guide_Name`.
```

这个不重要，可以在之后哪个 PR 顺手改一下就可以～

Reply:
好的~